src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayMethodTranslator.cs

```diff
@@ -198,6 +202,22 @@ static bool IsMappedToNonArray(SqlExpression arrayOrList)
                     inferredMapping);
             }
 
+            if (method.IsClosedFormOf(Enumerable_Remove))
+            {
+                var inferredMapping = ExpressionExtensions.InferTypeMapping(arrayOrList, arguments[0]);
```

Comment:
This is incorrect - `arguments[0]` is an item, whereas `arrayOrList` is an array, so they are different types, and we can't just infer directly (compare this to Concat just above, where both arguments are arrays).

See Enumerable_Append for a closer example to copy from (i.e. use `ApplyTypeMappingsOnItemAndArray)